# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xf86-input-mutouch/abi.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Version 1.2.1 doesn't build against the newest xorg because of an ABI change.

 - Peter Grasch, SIMON listens e.V.

diff -Naur xf86-input-mutouch-1.2.1/src/xf86MuTouch.c xf86-input-mutouch-mine/src/xf86MuTouch.c
--- xf86-input-mutouch-1.2.1/src/xf86MuTouch.c	2008-08-14 08:15:31.000000000 +0200
+++ xf86-input-mutouch-mine/src/xf86MuTouch.c	2009-11-28 10:33:12.698529247 +0100
@@ -749,6 +749,10 @@
   unsigned char		req[MuT_PACKET_SIZE];
   unsigned char		reply[MuT_BUFFER_SIZE];
   char			*id_string = DEVICE_ID(local->private_flags) == FINGER_ID ? "finger" : "stylus";
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+  Atom			labels[1] = { 0 };
+  Atom 			axes_labels[2] = { 0, 0 };
+#endif
 
   switch(mode) {
 
@@ -766,7 +770,11 @@
       /*
        * Device reports button press for up to 1 button.
        */
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+      if (InitButtonClassDeviceStruct(dev, 1, labels, map) == FALSE) {
+#else
       if (InitButtonClassDeviceStruct(dev, 1, map) == FALSE) {
+#endif
 	ErrorF("Unable to allocate ButtonClassDeviceStruct\n");
 	return !Success;
       }
@@ -779,6 +787,9 @@
        * screen to fit one meter.
        */
       if (InitValuatorClassDeviceStruct(dev, 2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+      					labels,
+#endif
 #if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
 					xf86GetMotionEvents,
 #endif
@@ -787,11 +798,19 @@
 	return !Success;
       }
       else {
-	InitValuatorAxisStruct(dev, 0, priv->min_x, priv->max_x,
+	InitValuatorAxisStruct(dev, 0,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+			       axes_labels[0],
+#endif
+			       priv->min_x, priv->max_x,
 			       9500,
 			       0     /* min_res */,
 			       9500  /* max_res */);
-	InitValuatorAxisStruct(dev, 1, priv->min_y, priv->max_y,
+	InitValuatorAxisStruct(dev, 1,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+			       axes_labels[1],
+#endif
+			       priv->min_y, priv->max_y,
 			       10500,
 			       0     /* min_res */,
 			       10500 /* max_res */);

